src/transaction/types/input.rs

```diff
@@ -247,9 +250,10 @@ impl Input {
     pub const fn input_owner(&self) -> Option<&Address> {
         match self {
             Self::CoinSigned { owner, .. } | Self::CoinPredicate { owner, .. } => Some(owner),
-            Self::MessageSigned { .. } | Self::MessagePredicate { .. } | Self::Contract { .. } => {
-                None
+            Self::MessageSigned { recipient, .. } | Self::MessagePredicate { recipient, .. } => {
```

Comment:
Same thing as Victor mentioned below, we shouldn't mix recipient and owner behind the same getter.